waku/v2/protocol/waku_store/protocol.nim

```diff
@@ -135,11 +135,46 @@ proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse {.gcsafe.}
   if queryRes.isErr():
     return HistoryResponse(messages: @[], pagingInfo: PagingInfo(), error: HistoryResponseError.INVALID_CURSOR)
 
-  let (messages, updatedPagingInfo) = queryRes.get()
+  let rows = queryRes.get()
   
+  if rows.len <= 0:
+    return HistoryResponse(messages: @[], error: HistoryResponseError.NONE)
+  
+  var messages = if rows.len <= int(qMaxPageSize): rows.mapIt(it[1])
+                 else: rows[0..^2].mapIt(it[1])
+  var pagingInfo = none(PagingInfo)
+
+  # The retrieved messages list should always be in chronological order
+  if not qAscendingOrder:
+    messages.reverse()
```

Comment:
Not the main benefit of moving paging logic to the protocol, but I like that the message stores don't have to take care of this protocol-specific idiosyncrasy. 👍